sfeir-open-source/talk-control
# WIP: Add node back-end

Closes #8 and #4 
### Expected behavior
The server should start and handle socket events.
### How to reproduce
- Start the server: `npm start`
- Connect any socketio client (exemple: [socketio client tool](https://amritb.github.io/socketio-client-tool/))
- Sending an init event with a slide array should initialize the state. The event should be formated like this : `socket.emit('init', {slides: [{"h": 0, "v": 0, "f": 0, "fMax":0}, {"h": 1, "v": 0, "f": 0, "fMax":0}, {"h": 1, "v": 1, "f": 0, "fMax":0}]})` where `fMax` is the number of fragments in the slide.
- Once the state is initialized you can send keyPressed events formated like this: `socket.emit('keyPressed', {key: 'arrowRight'})`. This should update the state and emit `'gotoSlide'` events.
